### PR TITLE
[MM-8404] Add an utility function to ignore channel-wide mentions on channels

### PR DESCRIPTION
**Pull request by letsila**
#### Summary
This is needed by this update on the mobile app https://github.com/mattermost/mattermost-mobile/pull/2302

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/9505

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by enahum** [comment]
@letsila thanks very much for this, can you please add some tests to this PR?

**Comment by letsila** [comment]
will do

File: src/utils/channel_utils.js [hunk 1]
```diff
@@ -556,6 +556,21 @@ export function isChannelMuted(member) {
     return member && member.notify_props ? (member.notify_props.mark_unread === 'mention') : false;
 }
 
+export function areChannelMentionsIgnored(channelMemberNotifyProps, currentUserNotifyProps) {
+    let ignoreChannelMentionsDefault = 'ignore_channel_mentions_off';
```
[the remark this answers is not in the document]
Comment (letsila) [reply src/utils/channel_utils.js]: done with the last commit :)

File: test/utils/channel_utils.test.js [hunk 2]
```diff
@@ -3,10 +3,11 @@
 
 import assert from 'assert';
 
-import General from 'constants/general';
+import {General, Preferences} from 'constants';
 import TestHelper from 'test/test_helper';
 
 import {canManageMembersOldPermissions, isAutoClosed, filterChannelsMatchingTerm} from 'utils/channel_utils';
+import {areChannelMentionsIgnored} from '../../src/utils/channel_utils';
```
Comment (enahum) [inline test/utils/channel_utils.test.js]: please change this path to `utils/channel_utils`

File: src/constants/preferences.js [hunk 3]
```diff
@@ -3,6 +3,10 @@
 // @flow
 
 export default {
+    IGNORE_CHANNEL_MENTIONS_ON: 'ignore_channel_mentions_on',
```
Comment (hmhealey) [inline src/constants/preferences.js]: Can you make a `src/constants/users.js` and move these into it? This file is specifically for the constants used for `Preference` objects

**Comment by jasonblais** [comment]
Hey @letsila - many thanks for your contribution here! I'm really looking forward this functionality myself 🎉 

Looks like we're really close to merging this! If you have any questions on the above two changes, let us know. Happy to help :)

**Comment by hmhealey** [comment]
I've merged this now so the other branch can be updated

**Pull request merged**